### PR TITLE
clnt-1420-gestures

### PR DESCRIPTION
**Pull request by atiupin**
https://qleanru.atlassian.net/browse/CLNT-1420

— Demo app
— Closing notification swiping up
— Individual config for each notification

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "qlean-react-native",
+  "rules": {
+    "react/prop-types": 0,
+    "react-native/no-inline-styles": 0,
+    "react-native/no-color-literals": 0
+  }
+}
```
Comment (ghost) [inline .eslintrc]: new line here and further

**Review by ghost — APPROVED** [review]

**Review by vadimshvetsov — APPROVED** [review]

File: MessageBar.js [hunk 2]
```diff
@@ -30,46 +37,68 @@ function Message({ message }) {
...
+  }
+  componentWillMount() {
+    this.panResponder = PanResponder.create({
+      onMoveShouldSetPanResponder: () => true,
+      onMoveShouldSetPanResponderCapture: () => true,
+      onPanResponderMove: (e, gesture) => {
+        if (
+          gesture.dy < -MIN_SWIPE_DISTANCE
+          && gesture.vy < -MIN_SWIPE_VELOCITY
+          && this.getConfig().closeOnSwipe
```
Comment (8of) [inline MessageBar.js]: Let's check this first.
If it's `false` - `gesture` params doesn't matter at all.

File: README.md [hunk 3]
```diff
@@ -14,25 +14,25 @@ Messages are not interactable in any way
...
 ## Installation
 ```
-$ npm install --save react-native-messages
+$ npm install react-native-messages
 ```
 
 ## Usage
-1. Add `<MessageBar/>` to the top of your view hierarchy, as a last component:
-```
+1. Add `<MessageBar/>` to the top of your view hierarchy, as a last component. If you are using `react-navigation`, root navigator should be in place of `<App/>`. Root view should have `flex: 1`:
```
Comment (8of) [inline README.md:21]: 👍  👍  👍

File: README.md [hunk 4]
```diff
@@ -61,4 +65,18 @@ function Message({ message }) {
...
+
+```
+$ git clone git@github.com:Qlean/react-native-messages.git
+$ cd react-native-messages/demo
+$ npm install
+$ npm start
+$ react-native run-android # assuming Android emulator is running: https://facebook.github.io/react-native/docs/getting-started.html#installing-dependencies
+```
+
+## Contibuting
```
Comment (8of) [inline README.md]: Let's fix the typo.

File: demo/ios/MessagesExample/AppDelegate.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
```
Comment (8of) [inline demo/ios/MessagesExample/AppDelegate.h]: Let's replace it with 2018-present, Qlean.

File: demo/ios/MessagesExample/AppDelegate.m [hunk 6]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
```
Comment (8of) [inline demo/ios/MessagesExample/AppDelegate.m]: Let's replace it with 2018-present, Qlean.

File: demo/ios/MessagesExample/main.m [hunk 7]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
```
Comment (8of) [inline demo/ios/MessagesExample/main.m]: Let's replace it with 2018-present, Qlean.

File: demo/ios/MessagesExampleTests/MessagesExampleTests.m [hunk 8]
```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
```
Comment (8of) [inline demo/ios/MessagesExampleTests/MessagesExampleTests.m]: Let's replace it with 2018-present, Qlean.

**Review by 8of — APPROVED** [review]

**Pull request merged**